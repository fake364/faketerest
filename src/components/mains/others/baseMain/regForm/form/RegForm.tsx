import React from 'react';
import Image from 'next/image';
import { useTranslation } from 'next-i18next';
import InputWithError from '../../../../../../common/components/inputs/commonInput/inputWithError/InputWithError';
import PrimaryButton from '../../../../../../common/components/buttons/primary-button/PrimaryButton';
import { useFormik } from 'formik';
import BaseButton from '../../../../../../common/components/buttons/base/BaseButton';
import IconButton from '../../../../../../common/components/buttons/icon-button/IconButton';
import { IconType } from '@react-icons/all-files';
import { FaFacebook } from '@react-icons/all-files/fa/FaFacebook';

type Props = {};

const RegForm: React.FC<Props> = () => {
  const { t } = useTranslation('main-page');
  const formik = useFormik({
    initialValues: {
      email: '',
      password: '',
      age: ''
    },
    onSubmit: () => {}
  });

  console.log(formik);

  return (
    <div className="p-[8px] bg-white flex w-[483px] min-w-[483px] items-center flex-col ml-[56px] rounded-form_radius">
      <div className="my-[16px]">
        <Image src={'/logo.png'} width={40} height={40} />
      </div>
      <div className="text-[36px] text-[#333333] font-semibold">
        {t('regForm.form.title')}
      </div>
      <div className="text-[16px] text-[#333333] mt-[8px]">
        {t('regForm.form.subtitle')}
      </div>
      <div className="flex flex-col w-[60%]">
        <InputWithError
          name="email"
          onChange={formik.handleChange}
          className="mt-[28px]"
          placeholder="Email"
        />
        <InputWithError
          name="password"
          onChange={formik.handleChange}
          type="password"
          className="mt-[8px]"
          placeholder="Create a password"
        />
        <InputWithError
          name="age"
          onChange={formik.handleChange}
          className="mt-[8px]"
          placeholder="Age"
        />
        <PrimaryButton
          className="mt-[8px] self-stretch"
          onClick={formik.handleSubmit}
        >
          Continue
        </PrimaryButton>
        <div className="mt-[8px] text-[#333333] font-bold">OR</div>
        <BaseButton
          className="mt-[8px] self-stretch bg-[#1877f2] rounded-common_radius py-[8px] text-white"
          onClick={formik.handleSubmit}
        >
          <div className="flex items-center justify-center">
            <FaFacebook className="mr-[8px] text-[24px]" />
            <div>Continue with facebook</div>
          </div>
        </BaseButton>
      </div>
    </div>
  );
};

export default RegForm;
