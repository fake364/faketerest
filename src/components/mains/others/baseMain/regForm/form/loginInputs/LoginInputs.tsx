import React, { useState } from 'react';
import { useFormik } from 'formik';
import axios from 'axios';
import sha256 from 'crypto-js/sha256';
import RegFormSpinner from '../spinner/RegFormSpinner';
import InputWithError from '../../../../../../../common/components/inputs/commonInput/inputWithError/InputWithError';
import PrimaryButton from '../../../../../../../common/components/buttons/primary-button/PrimaryButton';
import { useTranslation } from 'next-i18next';
import {
  getInitLoginFormValues,
  initializeLoginSchema
} from '../formValues/schemas/initializeLoginSchema';
import { loginFormNames } from '../formValues/schemas/GetLoginFormSchema';
import { StatusCodes } from 'http-status-codes';
import { AppDispatch } from '../../../../../../../redux/types';
import { useDispatch } from 'react-redux';
import { setIsLoggedIn } from '../../../../../../../redux/actions/metadata/actions';

type Props = {};

const LoginInputs: React.FC<Props> = () => {
  const { t: commonTranslations } = useTranslation('common');
  const dispatch: AppDispatch = useDispatch();
  const { t } = useTranslation('main-page');
  const { t: errors } = useTranslation('error-messages');
  const [error, setError] = useState<string>('');

  const formik = useFormik({
    initialValues: getInitLoginFormValues(),
    validationSchema: initializeLoginSchema(errors),
    validateOnChange: false,
    validateOnBlur: false,
    onSubmit: async (values) => {
      try {
        const response = await axios('/api/login', {
          responseType: 'json',
          method: 'POST',
          data: {
            [loginFormNames.password]: sha256(values.password).toString(),
            [loginFormNames.email]: values.email
          }
        });
        if (response.status === StatusCodes.OK) {
          dispatch(setIsLoggedIn(true));
          const header = document.querySelector('header');
          if (header) header.style.visibility = 'visible';
        }
      } catch (e) {
        console.error(e);
        if (e.response.status === StatusCodes.UNAUTHORIZED) {
          setError(errors('invalidAuth'));
        } else {
          setError(errors('somethingWentWrong'));
        }
      }
    }
  });

  return (
    <>
      {formik.isSubmitting && <RegFormSpinner />}

      <form className="flex flex-col w-full" onSubmit={formik.handleSubmit}>
        {error && (
          <label className="text-[#e60023] text-center px-[10px] block text-[14px] mt-[14px] mb-[14px]">
            {error}
          </label>
        )}
        <InputWithError
          name={loginFormNames.email}
          onChange={formik.handleChange}
          className="mt-[14px]"
          placeholder={commonTranslations('placeholders.email')}
          labelText={formik.errors[loginFormNames.email]}
        />
        <InputWithError
          name={loginFormNames.password}
          onChange={formik.handleChange}
          type="password"
          className="mt-[8px]"
          placeholder={commonTranslations('placeholders.password')}
          labelText={formik.errors[loginFormNames.password]}
        />
        <PrimaryButton className="mt-[8px] self-stretch" type="submit">
          {t('regForm.form.buttons.continue')}
        </PrimaryButton>
      </form>
    </>
  );
};

export default LoginInputs;
