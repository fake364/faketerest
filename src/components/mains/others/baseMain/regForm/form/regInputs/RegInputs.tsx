import React from 'react';
import InputWithError from '../../../../../../../common/components/inputs/commonInput/inputWithError/InputWithError';
import { regFormNames } from '../formValues/schemas/GetRegFormSchema';
import PrimaryButton from '../../../../../../../common/components/buttons/primary-button/PrimaryButton';
import { useTranslation } from 'next-i18next';
import { useFormik } from 'formik';
import {
  getInitRegFormValues,
  initializeRegSchema
} from '../formValues/schemas/initializeRegSchema';
import axios from 'axios';
import sha256 from 'crypto-js/sha256';
import { REGISTRATION_ERROR } from '../../../../../../../common/backend/models/constants/code';
import RegFormSpinner from '../spinner/RegFormSpinner';
import { StatusCodes } from 'http-status-codes';
import { setIsLoggedIn } from '../../../../../../../redux/actions/metadata/actions';
import { AppDispatch } from '../../../../../../../redux/types';
import { useDispatch } from 'react-redux';

export const RegInputs: React.FC = () => {
  const { t: commonTranslations } = useTranslation('common');
  const { t } = useTranslation('main-page');
  const { t: errors } = useTranslation('error-messages');
  const dispatch: AppDispatch = useDispatch();

  const formik = useFormik({
    initialValues: getInitRegFormValues(),
    validationSchema: initializeRegSchema(errors),
    validateOnChange: false,
    validateOnBlur: false,
    onSubmit: async (values) => {
      try {
        const response = await axios('/api/registration/create', {
          responseType: 'json',
          method: 'POST',
          data: {
            [regFormNames.password]: sha256(values.password).toString(),
            [regFormNames.age]: values.age,
            [regFormNames.email]: values.email
          }
        });
        if (response.status === StatusCodes.OK) {
          dispatch(setIsLoggedIn(true));
          const header = document.querySelector('header');
          if (header) header.style.visibility = 'visible';
        }
      } catch (e) {
        console.log(e);
        if (e.response.data.errorCode === REGISTRATION_ERROR.email) {
          formik.setFieldError(regFormNames.email, errors('existentEmail'));
        }
      }
    }
  });

  return (
    <>
      {formik.isSubmitting && <RegFormSpinner />}
      <form className="flex flex-col w-full" onSubmit={formik.handleSubmit}>
        <InputWithError
          name={regFormNames.email}
          onChange={formik.handleChange}
          className="mt-[28px]"
          placeholder={commonTranslations('placeholders.email')}
          labelText={formik.errors[regFormNames.email]}
        />
        <InputWithError
          name={regFormNames.password}
          onChange={formik.handleChange}
          type="password"
          className="mt-[8px]"
          placeholder={commonTranslations('placeholders.createPassword')}
          labelText={formik.errors[regFormNames.password]}
        />
        <InputWithError
          name={regFormNames.age}
          onChange={formik.handleChange}
          className="mt-[8px]"
          placeholder={commonTranslations('placeholders.age')}
          labelText={formik.errors[regFormNames.age]}
        />
        <PrimaryButton className="mt-[8px] self-stretch" type="submit">
          {t('regForm.form.buttons.continue')}
        </PrimaryButton>
      </form>
    </>
  );
};
