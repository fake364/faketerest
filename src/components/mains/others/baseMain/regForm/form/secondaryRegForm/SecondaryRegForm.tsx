import React from 'react';
import Link from 'next/link';
import { useTranslation } from 'next-i18next';

type Props = { onTriggerFormMode?: () => void; isRegistrationForm: boolean };

const SecondaryRegForm: React.FC<Props> = ({
  onTriggerFormMode,
  isRegistrationForm
}) => {
  const { t } = useTranslation('main-page');

  return (
    <>
      <div className="mt-[14px] text-center text-[12px] text-gray-600">
        {t('regForm.form.terms.youAgreeWith')}{' '}
        <span className="font-bold whitespace-nowrap">
          <Link href={'/tos'}>{t('regForm.form.terms.termsOfService')}</Link>
        </span>{' '}
        {t('regForm.form.terms.youVeRead')}{' '}
        <span className="font-bold whitespace-nowrap">
          <Link href={'/privacy'}>{t('regForm.form.terms.privacyPolicy')}</Link>
        </span>
      </div>
      {onTriggerFormMode && (
        <div
          className="mt-[12px] text-[13px] text-center text-gray-600 cursor-pointer font-semibold"
          onClick={onTriggerFormMode}
        >
          {!isRegistrationForm
            ? t('regForm.form.registerLink')
            : t('regForm.form.logInLink')}
        </div>
      )}
      <div className=" mt-[10px] text-[13px] text-center text-gray-600 font-semibold pb-[14px]">
        {t('regForm.form.areYouABusiness')}{' '}
        <span>
          <Link href={'/business/create'}>
            {t('regForm.form.getStartedHere')}
          </Link>
        </span>
      </div>
    </>
  );
};

export default SecondaryRegForm;
