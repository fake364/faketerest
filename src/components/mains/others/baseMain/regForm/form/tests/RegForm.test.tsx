import React from 'react';
import { shallow } from 'enzyme';
import RegForm from '../RegForm';
import { RegInputs } from '../regInputs/RegInputs';
import SecondaryRegForm from '../secondaryRegForm/SecondaryRegForm';
import LoginInputs from '../loginInputs/LoginInputs';

describe('RegForm', () => {
  it('should render Reg form if isRegShown true', () => {
    const wrapper = shallow(<RegForm isRegistrationShown={true} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render login form is isRegShown false', () => {
    const wrapper = shallow(<RegForm isRegistrationShown={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should switch regform to login form mode on trigger button callback', () => {
    let isRegistrationShown = true;
    let changeForm = jest.fn((flag: boolean) => (isRegistrationShown = flag));
    const wrapper = shallow(
      <RegForm
        isRegistrationShown={isRegistrationShown}
        setRegistrationShown={changeForm}
      />
    );
    expect(wrapper.find(RegInputs)).toHaveLength(1);
    wrapper.find(SecondaryRegForm).props().onTriggerFormMode();
    wrapper.setProps({
      isRegistrationShown: false,
      setRegistrationShown: changeForm
    });
    wrapper.update();
    expect(changeForm).toHaveBeenCalled();
    expect(wrapper.find(LoginInputs)).toHaveLength(1);
  });
});
