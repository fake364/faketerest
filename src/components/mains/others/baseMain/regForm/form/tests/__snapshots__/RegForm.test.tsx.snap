// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegForm should render Reg form if isRegShown true 1`] = `
<div
  className="bg-white flex w-[483px] min-w-[483px]
 items-center flex-col ml-[56px] rounded-form_radius overflow-hidden
 shadow-[rgb(0_0_0/45%)_0px_2px_10px] relative"
>
  <RegFormHeader />
  <div
    className="flex flex-col w-[60%]"
  >
    <RegInputs />
    <div
      className="mt-[8px] text-[#333333] font-bold text-center"
    >
      regForm.form.buttons.or
    </div>
    <IconTextButton
      Icon={[Function]}
      className="bg-[#1877f2] text-white"
      onClick={null}
      text="regForm.form.buttons.continueWithFacebook"
    />
    <IconTextButton
      Icon={[Function]}
      className="text-[#3c4043] border-[1px] border-[#dadce0] text-[15px] font-[500] mt-[12px]"
      onClick={null}
      text="regForm.form.buttons.continueWithGoogle"
    />
    <SecondaryRegForm
      isRegistrationForm={true}
      onTriggerFormMode={null}
    />
  </div>
</div>
`;

exports[`RegForm should render login form is isRegShown false 1`] = `
<div
  className="bg-white flex w-[483px] min-w-[483px]
 items-center flex-col ml-[56px] rounded-form_radius overflow-hidden
 shadow-[rgb(0_0_0/45%)_0px_2px_10px] relative"
>
  <RegFormHeader />
  <div
    className="flex flex-col w-[60%]"
  >
    <LoginInputs />
    <div
      className="mt-[8px] text-[#333333] font-bold text-center"
    >
      regForm.form.buttons.or
    </div>
    <IconTextButton
      Icon={[Function]}
      className="bg-[#1877f2] text-white"
      onClick={null}
      text="regForm.form.buttons.continueWithFacebook"
    />
    <IconTextButton
      Icon={[Function]}
      className="text-[#3c4043] border-[1px] border-[#dadce0] text-[15px] font-[500] mt-[12px]"
      onClick={null}
      text="regForm.form.buttons.continueWithGoogle"
    />
    <SecondaryRegForm
      isRegistrationForm={false}
      onTriggerFormMode={null}
    />
  </div>
</div>
`;
